classes/observation_manager.php

```diff
@@ -296,4 +298,96 @@ public static function reorder_observation_point(int $observationid, int $obpoin
+        ON pts.id = sess_resp.obs_pt_id WHERE pts.obs_id = :observationid;';
+
+        $pointsandresponses = $DB->get_records_sql($sql, ['observationid' => $observationid, 'sessionid' => $sessionid]);
+
+        return $pointsandresponses;
+    }
+
+    /**
+     * Submits a response to a particular observation point for a given sesssion.
+     * @param int $sessionid ID of the observation session
+     * @param int $pointid ID of the observation point this response is for
+     * @param mixed $data data object returned from the pointmarking_form
+     */
+    public static function submit_point_response(int $sessionid, int $pointid, $data) {
+        global $DB;
+
+        if ($data->grade_given < 0 || !is_int($data->grade_given)) {
+            throw new \coding_exception("Grade given must be an integer that is zero or more");
+        }
+
+        if (!isset($data->response)) {
+            throw new \coding_exception("No response was found in the data.");
+        }
+
+        $sessioninfo = \mod_observation\session_manager::get_session_info($sessionid);
+        $pointdata = self::get_existing_point_data($sessioninfo['obid'], $pointid);
+        if ($data->grade_given > $pointdata->max_grade) {
+            throw new \coding_exception("Grade given must be less than the max grade.");
+        }
+
+        // See if a response already exists for this session and pointid.
+        $existingresponse = $DB->get_record('observation_point_responses', ['obs_pt_id' => $pointid, 'obs_ses_id' => $sessionid]);
+
+        // Clean data.
+        $dbdata = [
+            'obs_pt_id' => $pointid,
+            'obs_ses_id' => $sessionid,
+            'grade_given' => $data->grade_given,
+            'response' => $data->response,
+            'ex_comment' => $data->ex_comment,
+        ];
+
+        if ($existingresponse === false) {
+            // Insert new.
+            $DB->insert_record('observation_point_responses', $dbdata);
+        } else {
+            // Update existing.
+            $dbdata['id'] = $existingresponse->id;
+            $DB->update_record('observation_point_responses', $dbdata);
+        }
+    }
+
+    /**
+     * Generates a HTML table that summarises the observation points and their responses
+     * @param int $observationid ID of the observation instance
+     * @param int $sessionid ID of the observation session
+     * @return string HTML string
+     */
+    public static function format_points_and_responses($observationid, $sessionid) {
+        $pointsandresponses = self::get_points_and_responses($observationid, $sessionid);
+
+        $table = new \html_table();
+        $table->head = ['Title', 'Response', 'Grade Given'];
+
+        $table->data = array_map(function($item) {
+            return [
+                $item->title,
+                $item->response,
+                $item->grade_given,
+            ];
+        }, $pointsandresponses);
+
+        return html_writer::table($table);
```

Comment:
this can likely just be \html_writer and kill the include at the top of the file